### ISSUE TITLE
ss.Births is not CRN safe

### ISSUE THREAD

**Issue by cliffckerr** [starsimhub/starsim#404]

**Comment by daniel-klein** [comment]
Yes! Births are deterministic in Starsim, but the properties of agents added to the simulation may not achieve CRN (still not sure). The two key properties of `people` are `slot` and `female`.

* Slots default to UID
* Female is drawn from `ss.bernoulli(p=0.5)`

Instead of deterministically choosing the number of new agents to add to the sim, we could do something very similar to `Pregnancy`, just not create embryos nor the maternal network. This would also provide the benefit of giving each new agent a biological mother.

**Comment by kaa2102** [comment]
@daniel-klein Do we want to create both slots and draws from the bernouli dist'n, or just the latter?

**Comment by daniel-klein** [comment]
Note that with the addition of sc.randround in the latest #724, `Births` is now quite far from CRN safe. CRN seekers will need to use `Pregnancy` instead. We still need a simple CRN-safe version of Pregnancy, perhaps to compliment a performance optimized non-CRN implementation. cc @cliffckerr

**Comment by RomeshA** [comment]
This was a bit of a gotcha for us, since we wanted just simple demographics and high performance we used `Births` instead of `Pregnancy` but having CRN is a big part of the appeal of using the Starsim platform, I think it's reasonably important that our default simple demographic solution should be more or less CRN-safe

**Comment by cliffckerr** [comment]
I think most of the slowness from `Pregnancy` is from the CRN-ness, though -- specifically the slot scale:
```py
import sciris as sc
import starsim as ss

pars = dict(n_agents=100e3, networks='random', diseases='sis', verbose=0)

s1 = ss.Sim(label='births', **pars, demographics=ss.Births())
s2 = ss.Sim(label='pregnancy', **pars, demographics=ss.Pregnancy())
s3 = ss.Sim(label='preg_low_slot', **pars, demographics=ss.Pregnancy(slot_scale=1.1))

for sim in [s1, s2, s3]:
    with sc.timer(sim.label):
        sim.run()

births: 3.20 s
pregnancy: 3.92 s
preg_low_slot: 3.02 s
```

**Comment by cliffckerr** [comment]
Closed for real by https://github.com/starsimhub/starsim/pull/1012! Births even assign parents now.
